certbot-apache/certbot_apache/dualparser.py

```diff
@@ -0,0 +1,266 @@
+        object encapsulating both of the newly created objects """
+
+        primary_new = self.primary.add_child_comment(comment, position)
+        secondary_new = self.secondary.add_child_comment(comment, position)
+        self.assertEqual(primary_new, secondary_new)
+        new_comment = DualCommentNode(primary=primary_new, secondary=secondary_new)
+        return new_comment
+
+    def find_blocks(self, name, exclude=True):  # pragma: no cover
+        """
+        Performs a search for BlockNodes using both implementations and
+        checks the results. This is built upon the assumption that unimplemented
+        find_* methods return a list with a single assertion passing object.
+        After the assertion, it creates a list of newly created DualBlockNode
+        instances that encapsulate the pairs of returned BlockNode objects.
+        """
+
+        primary_blocks = self.primary.find_blocks(name, exclude)
+        secondary_blocks = self.secondary.find_blocks(name, exclude)
+
+        new_blocks = list()
+        # Check for assertion pass for results, as zip loses data otherwise.
+        # unimplemented find_* methods should return a single object with
+        # assertion pass.
+        if len(primary_blocks) != len(secondary_blocks):
+            if secondary_blocks:
+                if self.isPass(secondary_blocks[0]):
+                    for c in primary_blocks:
+                        new_blocks.append(DualBlockNode(
+                            primary=c, secondary=secondary_blocks[0]))
+                else:
+                    # Break as we got confusing results.
+                    assert False
+            else:
+                # Break as we got confusing results.
+                assert False
+        else:
+            for twins in zip(primary_blocks, secondary_blocks):
+                self.assertEqual(twins[0], twins[1])
+                new_blocks.append(DualBlockNode(primary=twins[0],
+                                                secondary=twins[1]))
+        return new_blocks
+
+    def find_directives(self, name, exclude=True):  # pragma: no cover
+        """
+        Performs a search for DirectiveNodes using both implementations and
+        checks the results. This is built upon the assumption that unimplemented
+        find_* methods return a list with a single assertion passing object.
+        After the assertion, it creates a list of newly created DualDirectiveNode
+        instances that encapsulate the pairs of returned DirectiveNode objects.
+        """
+        primary_directives = self.primary.find_directives(name, exclude)
+        secondary_directives = self.secondary.find_directives(name, exclude)
+
+        new_dirs = list()
+        # Check for assertion pass for results, as zip loses data otherwise.
+        # unimplemented find_* methods should return a single object with
+        # assertion pass.
+        if len(primary_directives) != len(secondary_directives):
```

Comment:
nit: I think this `if/else` block can be simplified a lot and maybe even shared between different `find_*` implementations.